### ISSUE TITLE
gitgutter stopped working on some repos starting from 8345c35770ffc6fc4088c36406d1e24170aabcc6

### ISSUE THREAD

**Issue by mcagl** [airblade/vim-gitgutter#245]
Hello,
it's some time that I noticed that gitgutter stopped working on some of my git repositories (mainly small to big django projects, which are unfortunately the ones where gitgutter is really important :-) )

For example in my git repository for ~/.vim it keeps working (and here it's, well, less fundamental)

I did (manually) something that I called "the poor man's git bisect", and found that the exact commit that breaks gitgutter for me is 8345c35770ffc6fc4088c36406d1e24170aabcc6

I maintain my ~/.vim directory here on github, if it's useful for you to debug the situation: https://github.com/mcagl/dotvim

I tried to disable 90% of my plugins, and it didn't solve the problem. Doing 'git checkout 1a7f1c5bf64726c338d82d62f9be64dcfd289a09' inside ~/.vim/bundle/vim-gitgutter did (which is the commit just before the one that breaks (for my use cases) the functionality).

If gitgutter is checked to a commit after that, it stops showing the signs, it stops updating the lines count in my airline, etcetera.

Please tell me if I can be more useful, thanks in advance.

**Comment by airblade** [comment]
Thanks for pinpointing the problematic commit: that's really helpful.

That commit is supposed to stop gitgutter from running on any files inside git's own `.git/` directory.  It decides whether a file is inside a `.git/` directory by matching the file's path against `\.git\([/\\].*\)\?$`.

Please could you go to one of your projects where vim-gitgutter has stopped working, open one of the files in vim, and tell me what `:echo expand('%:p:h')` gives?

**Comment by airblade** [comment]
Hmm, I think a better regex would be: `[/\\]\.git\($\|[/\\]\)`...

**Comment by mcagl** [comment]
Thank you for the response to the issue.

Gotcha!!!
The problem is, then, with the names of my directories (so it's my fault, sorry to bother you).

If I have a project called, let's say, 'foo', in a git repository, I call the directory containing it 'foo.git' (that's because I have some (lame, I know..) scripts written by me to update repositories, that rely on the directory name to tell the used VCS: foo.git, foo.svn, foo.hg, etc etc)

The output of the command (given after checking out gitgutter to `master`) tells the directory containing the file I opened (I opened `/home/mark/projects/foo.git/foo/views.py`), like this: `/home/mark/projects/foo.git/foo`

**Comment by airblade** [comment]
No, it's not your fault at all – it's my lamentable regular expression.

Please could you replace the [existing one](https://github.com/airblade/vim-gitgutter/blob/e098491906c4e6d8a2a4d10ab60b91570eb81113/autoload/gitgutter/utility.vim#L19) with the following line and see if it works?

```viml
  return gitgutter#utility#full_path_to_directory_of_file() !~ '[/\\]\.git\($\|[/\\]\)'
```

**Comment by mcagl** [comment]
I tried the regexp in vim-gitgutter @ master and it works now (as a double check I tried the two regexes back and forth and it seems to solve the problem for my use cases).

Thank you, really!

**Comment by airblade** [comment]
The fix is on master now (937930e).  Thanks for your help with this!
